otx/algorithms/segmentation/adapters/mmseg/models/heads/custom_fcn_head.py

```diff
@@ -4,30 +4,138 @@
+from mmseg.models.losses import accuracy
+from mmseg.ops import resize
+from torch import nn
 
-from .mixin import (
-    AggregatorMixin,
-    MixLossMixin,
-    PixelWeightsMixin2,
-    SegmentOutNormMixin,
+from otx.algorithms.segmentation.adapters.mmseg.models.utils import IterativeAggregator
+from otx.algorithms.segmentation.adapters.mmseg.utils import (
+    get_valid_label_mask_per_batch,
 )
 
 
 @HEADS.register_module()
-class CustomFCNHead(
-    SegmentOutNormMixin, AggregatorMixin, MixLossMixin, PixelWeightsMixin2, FCNHead
-):  # pylint: disable=too-many-ancestors
+class CustomFCNHead(FCNHead):  # pylint: disable=too-many-ancestors
     """Custom Fully Convolution Networks for Semantic Segmentation."""
 
-    def __init__(self, *args, **kwargs):
+    def __init__(
+        self,
+        enable_aggregator=False,
+        aggregator_min_channels=None,
+        aggregator_merge_norm=None,
+        aggregator_use_concat=False,
+        *args,
+        **kwargs
+    ):
+
+        in_channels = kwargs.get("in_channels")
+        in_index = kwargs.get("in_index")
+        norm_cfg = kwargs.get("norm_cfg")
+        conv_cfg = kwargs.get("conv_cfg")
+        input_transform = kwargs.get("input_transform")
+
+        aggregator = None
+        if enable_aggregator:  # Lite-HRNet aggregator
+            assert isinstance(in_channels, (tuple, list))
+            assert len(in_channels) > 1
+
+            aggregator = IterativeAggregator(
+                in_channels=in_channels,
+                min_channels=aggregator_min_channels,
+                conv_cfg=conv_cfg,
+                norm_cfg=norm_cfg,
+                merge_norm=aggregator_merge_norm,
+                use_concat=aggregator_use_concat,
+            )
+
+            aggregator_min_channels = aggregator_min_channels if aggregator_min_channels is not None else 0
+            # change arguments temporarily
+            kwargs["in_channels"] = max(in_channels[0], aggregator_min_channels)
+            kwargs["input_transform"] = None
+            if in_index is not None:
+                kwargs["in_index"] = in_index[0]
+
         super().__init__(*args, **kwargs)
 
+        self.aggregator = aggregator
+
+        # re-define variables
+        self.in_channels = in_channels
+        self.input_transform = input_transform
+        self.in_index = in_index
+
+        self.ignore_index = 255
+
         # get rid of last activation of convs module
         if self.act_cfg:
             self.convs[-1].with_activation = False
             delattr(self.convs[-1], "activate")
 
         if kwargs.get("init_cfg", {}):
             self.init_weights()
+
+    def _transform_inputs(self, inputs):
+        if self.aggregator is not None:
+            inputs = self.aggregator(inputs)[0]
+        else:
+            inputs = super()._transform_inputs(inputs)
+
+        return inputs
+
+    def forward_train(self, inputs, img_metas, gt_semantic_seg, train_cfg):
```

Comment:
It would be much better if all functions have own type hint!

Reply:
Thanks for comments. I added.